### ISSUE TITLE
Add a `filegroup`-like rule for haskell targets

### ISSUE THREAD

**Issue by regnat** [tweag/rules_haskell#753]
While refactoring a codebase, I tried to internally split some packages while keeping the same interface, *i.e* structure them as follows:

```bzl
haskell_library(
  name = "sub-lib1",
  …,
  visibility = ["//visibility:private"],
)

haskell_library(
  name = "sub-lib2",
  …,
  visibility = ["//visibility:private"],
)

haskell_library(
  name = "sub-lib3",
  …,
  visibility = ["//visibility:private"],
)

filegroup(
  name = "mylib",
  srcs = [ ":sub-lib1", ":sub-lib2", ":sub-lib3" ],
  visibility = ["//visiblity:public"],
)
```

so that the dependencies which used to depend on `//mylib:mylib` can continue to do so and the split remains internal.

However using a `filegroup` for that doesn't work since it won't forward the `HaskellLibraryInfo` providers and thus `//mylib:mylib` won't be recognized as a set of haskell deps and will be ignored.

This led me to the idea of a `haskell_target_group` rule which would combine a set of `haskell_library`/`haskell_binary` targets (meaning properly combine the `HaskellLibrary`/`HaskellBinary` providers so that it looks like a single target from the outside)

In addition to the pattern above, this would give "for free" a multi-targets repl rule (actually this would probably borrow a lot of the implementation from #736 and make the actual repl rule just a thin layer on top of it)

**Comment by Profpatsch** [comment]
cc @aherrmann

**Comment by aherrmann** [comment]
>In addition to the pattern above, this would give "for free" a multi-targets repl rule (actually this would probably borrow a lot of the implementation from #736 and make the actual repl rule just a thin layer on top of it)

The multi target REPL in #736 goes a bit further in that it can also load targets from the local workspace by source that are not explicitly listed in `deps`. In the context of `haskell_target_group` that means that it can also load targets from source that are not listed in `haskell_target_group`, but are a transitive dependency of one of the listed packages. This is very useful as it allows you to also quickly reload source changes of transitive dependencies in the REPL.

**Comment by regnat** [comment]
> The multi target REPL in #736 goes a bit further in that it can also load targets from the local workspace by source that are not explicitly listed in deps.

Oh didn't see that (admittedly I didn't look closely at it ;) ) That's awesome! (and indeed wouldn't be handled by `haskell_group` directly)

**Comment by mboes** [comment]
This was fixed in #1008.
